[material-ui] @mui/material-nextjs v5.15.4 is broken

### Search keywords

@mui/material-nextjs import ts error

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

#### Obvious difference in file structure between versions
| 5.15.4 | 5.15.3 |
| - | - |
| ![image](https://github.com/mui/material-ui/assets/47712901/2bd5237d-78f3-439a-9cb8-520af2690c3a) | ![image](https://github.com/mui/material-ui/assets/47712901/b9813653-8a6b-4c7f-932f-156c8342fc2e) |


Steps:
1. Install latest version of `@mui/material-nextjs`
2. Try importing anything, e.g. `import { AppRouterCacheProvider } from "@mui/material-nextjs/v14-appRouter";`
3. Get the error.


### Current behavior

When trying to import `@mui/material-nextjs/v14-appRouter` (or any other module) getting an error message that says `Cannot find module '@mui/material-nextjs/v14-appRouter' or its corresponding type declarations.`.

### Expected behavior

Imports are working normally without any issues.

### Context

I've encountered this error when simply following official documentation. This must be a package build/publishing error.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: macOS 12.6
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
    pnpm: 8.6.10 - ~/.nvm/versions/node/v18.16.1/bin/pnpm
  Browsers:
    Chrome: 120.0.6099.199
    Edge: Not Found
    Safari: 16.1
  npmPackages:
    @emotion/react: ^11.11.3 => 11.11.3 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.31 
    @mui/core-downloads-tracker:  5.15.4 
    @mui/icons-material: ^5.15.4 => 5.15.4 
    @mui/material: ^5.15.4 => 5.15.4 
    @mui/material-nextjs: 5.15.3 => 5.15.3 
    @mui/private-theming:  5.15.4 
    @mui/styled-engine:  5.15.4 
    @mui/system:  5.15.4 
    @mui/types:  7.2.13 
    @mui/utils:  5.15.4 
    @types/react: ^18.2.37 => 18.2.37 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ^5.1.6 => 5.1.6 
```
</details>

--- a/packages/mui-material-nextjs/package.json
+++ b/packages/mui-material-nextjs/package.json
@@ -32,7 +32,7 @@
     "build:copy-files": "node ../../scripts/copyFiles.mjs",
     "build:types": "node ../../scripts/buildTypes.mjs",
     "prebuild": "rimraf build tsconfig.build.tsbuildinfo",
-    "release": "pnpm build && npm publish build",
+    "release": "pnpm build && pnpm publish",
     "test": "cd ../../ && cross-env NODE_ENV=test mocha 'packages/mui-utils/**/*.test.{js,ts,tsx}'",
     "typescript": "tsc -p tsconfig.json"
   },
@@ -65,7 +65,8 @@
   },
   "sideEffects": false,
   "publishConfig": {
-    "access": "public"
+    "access": "public",
+    "directory": "build"
   },
   "engines": {
     "node": ">=12.0.0"

--- a/packages/mui-material-nextjs/package.json
+++ b/packages/mui-material-nextjs/package.json
@@ -33,7 +33,7 @@
     "build:types": "node ../../scripts/buildTypes.mjs",
     "prebuild": "rimraf build tsconfig.build.tsbuildinfo",
     "release": "pnpm build && pnpm publish",
-    "test": "cd ../../ && cross-env NODE_ENV=test mocha 'packages/mui-utils/**/*.test.{js,ts,tsx}'",
+    "test": "cd ../../ && cross-env NODE_ENV=test mocha 'packages/mui-material-nextjs/**/*.test.{js,ts,tsx}'",
     "typescript": "tsc -p tsconfig.json"
   },
   "devDependencies": {

